pkg/cmd/admin/migrate/images/images.go

```diff
@@ -0,0 +1,488 @@
+	if err != nil {
+		return err
+	}
+	o.Client = osclient
+
+	return nil
+}
+
+func (o *MigrateImageReferenceOptions) Validate() error {
+	if len(o.Mappings) == 0 {
+		return fmt.Errorf("at least one mapping argument must be specified: REGISTRY/NAME=REGISTRY/NAME")
+	}
+	return o.ResourceOptions.Validate()
+}
+
+func (o *MigrateImageReferenceOptions) Run() error {
+	return o.ResourceOptions.Visit(func(info *resource.Info) (migrate.Reporter, error) {
+		return o.transform(info.Object)
+	})
+}
+
+// save invokes the API to alter an object. The reporter passed to this method is the same returned by
+// the migration visitor method (for this type, transformImageReferences). It should return an error
+// if the input type cannot be saved. It returns migrate.ErrRecalculate if migration should be re-run
+// on the provided object.
+func (o *MigrateImageReferenceOptions) save(info *resource.Info, reporter migrate.Reporter) error {
+	switch t := info.Object.(type) {
+	case *imageapi.ImageStream:
+		if reporter.(imageChangeInfo).status {
+			updated, err := o.Client.ImageStreams(t.Namespace).UpdateStatus(t)
+			if err != nil {
+				return migrate.DefaultRetriable(info, err)
+			}
+			info.Refresh(updated, true)
+			return migrate.ErrRecalculate
+		}
+		if reporter.(imageChangeInfo).spec {
+			updated, err := o.Client.ImageStreams(t.Namespace).Update(t)
+			if err != nil {
+				return migrate.DefaultRetriable(info, err)
+			}
+			info.Refresh(updated, true)
+		}
+		return nil
+	default:
+		if _, err := resource.NewHelper(info.Client, info.Mapping).Replace(info.Namespace, info.Name, false, info.Object); err != nil {
+			return migrate.DefaultRetriable(info, err)
+		}
+		//return fmt.Errorf("resource %q does not have a save method implemented (%T)", info.Mapping.Resource, t)
+	}
+	return nil
+}
+
+// transform checks image references on the provided object and returns either a reporter (indicating
+// that the object was recognized and whether it was updated) or an error.
+func (o *MigrateImageReferenceOptions) transform(obj runtime.Object) (migrate.Reporter, error) {
+	fn := o.Mappings.MapReference
+	switch t := obj.(type) {
+	case *imageapi.Image:
+		var changed bool
+		if updated := fn(t.DockerImageReference); updated != t.DockerImageReference {
+			changed = true
+			t.DockerImageReference = updated
+		}
+		return reporter(changed), nil
+	case *imageapi.ImageStream:
+		var info imageChangeInfo
+		if len(t.Spec.DockerImageRepository) > 0 {
+			info.spec = updateString(&t.Spec.DockerImageRepository, fn)
+		}
+		for _, ref := range t.Spec.Tags {
+			if ref.From == nil || ref.From.Kind != "DockerImage" {
+				continue
+			}
+			info.spec = updateString(&ref.From.Name, fn) || info.spec
+		}
+		for _, events := range t.Status.Tags {
+			for i := range events.Items {
+				info.status = updateString(&events.Items[i].DockerImageReference, fn) || info.status
+			}
+		}
+		return info, nil
+	case *kapi.Secret:
+		switch t.Type {
+		case kapi.SecretTypeDockercfg:
```

Comment:
I don't want to exclude system maintained ones in this case (after our discussion) because admins may want to migrate for other reasons.